python/lsst/daf/butler/time_utils.py

```diff
@@ -136,10 +177,10 @@ def nsec_to_astropy(self, time_nsec: int) -> astropy.time.Time:
         that the number falls in the supported range and can produce output
         time that is outside of that range.
         """
-        jd1, jd2 = divmod(time_nsec, self._NSEC_PER_DAY)
-        delta = astropy.time.TimeDelta(float(jd1), float(jd2) / self._NSEC_PER_DAY, format="jd", scale="tai")
-        value = self.epoch + delta
-        return value
+        jd1, jd2 = divmod(time_nsec, 1_000_000_000)
+        return astropy.time.Time(
+            float(jd1), jd2 / 1_000_000_000, format="unix_tai_fast", scale="tai", precision=6
```

Comment:
@andy-slac  We have to pass in seconds here for jd1 rather than days (and so require the divmod call again) because we are subclassing `TimeUnixTai`?

Reply:
Right, I decided that we want to use TimeUnixTai instead of TimeJD format as a base because this is exactly what we do, and TimeUnix takes seconds. It would still be possible to pass days here, but I was worried it could cause confusion somewhere else.